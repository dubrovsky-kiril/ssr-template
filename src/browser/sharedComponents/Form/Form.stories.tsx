import * as React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import Form from "./Form";
import Input from "../Input/Input";

storiesOf("Form", module).add(
  "default",
  withInfo()(() => (
    <Form formId="productIdForm" handleOnSubmit={() => {}} buttonTxt="Search">
      <Input id="productId" label="Product id" type="text" />
    </Form>
  ))
);
