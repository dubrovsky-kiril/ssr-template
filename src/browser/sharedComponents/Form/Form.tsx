import * as React from "react";
import Button from "../Button/Button";

type FormTypes = {
  /** connects form with submit button */
  formId: string;
  /** should contain inputs, selectors or other data entrance elements  */
  children: JSX.Element;
  /** should contain submit handler function that will receive children elements data */
  handleOnSubmit(event: React.FormEvent<HTMLElement>): void;
  /** should contain button text */
  buttonTxt: string;
};

const Form: React.FunctionComponent<FormTypes> = ({
  formId,
  children,
  handleOnSubmit,
  buttonTxt
}) => (
  <form id={formId} onSubmit={handleOnSubmit}>
    <div>{children}</div>
    <Button type="submit" form={formId} txt={buttonTxt} />
  </form>
);

export default Form;
