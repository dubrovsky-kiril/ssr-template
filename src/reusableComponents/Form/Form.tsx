import * as React from "react";
import Button from "../Button/Button";

type FormTypes = {
  /** connects form with submit button */
  formId: string;
  /** should contain inputs, selectors or other data entrance elements  */
  children: JSX.Element;
  /** should contain submit handler function that will receive children elements data */
  onSubmit(event: React.FormEvent<HTMLElement>): void;
  /** should contain button text */
  buttonTxt: string;
};

const Form: React.FunctionComponent<FormTypes> = ({
  formId,
  children,
  onSubmit,
  buttonTxt
}) => (
  <form id={formId} onSubmit={onSubmit}>
    <div>{children}</div>
    <Button type="submit" form={formId} txt={buttonTxt} />
  </form>
);

export default Form;
