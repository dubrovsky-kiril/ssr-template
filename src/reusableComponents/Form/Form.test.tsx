import * as React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Form from "./Form";

describe("Form", () => {
  it("should apply 'formId' prop to form and Button", () => {
    const wrapper = shallow(
      <Form formId="productIdForm" onSubmit={() => {}} buttonTxt="Search">
        <input />
      </Form>
    );

    expect(wrapper.find("form").props().id).toBe("productIdForm");
    expect(wrapper.find("Button").props().form).toBe("productIdForm");
  });
  it("should contain passed children ", () => {
    const children = <input />;
    const wrapper = shallow(
      <Form formId="productIdForm" onSubmit={() => {}} buttonTxt="Search">
        {children}
      </Form>
    );

    expect(
      wrapper
        .find("div")
        .children()
        .contains(children)
    ).toEqual(true);
  });
  it("should apply onSubmit handler", () => {
    const handleOnSubmit = () => {};
    const wrapper = shallow(
      <Form formId="productIdForm" onSubmit={handleOnSubmit} buttonTxt="Search">
        <input />
      </Form>
    );

    expect(wrapper.find("form").props().onSubmit).toBe(handleOnSubmit);
  });
  it("should apply 'buttonTxt' prop to Button", () => {
    const wrapper = mount(
      <Form formId="productIdForm" onSubmit={() => {}} buttonTxt="Search">
        <input />
      </Form>
    );

    expect(wrapper.find("button").text()).toBe("Search");
    wrapper.unmount();
  });
  it("should match snapshot", () => {
    const tree = shallow(
      <Form formId="productIdForm" onSubmit={() => {}} buttonTxt="Search">
        <input />
      </Form>
    );

    expect(toJson(tree)).toMatchSnapshot();
  });
});
