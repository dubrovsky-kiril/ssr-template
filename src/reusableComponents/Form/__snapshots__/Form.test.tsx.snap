// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot 1`] = `
<form
  id="productIdForm"
  onSubmit={[Function]}
>
  <div>
    <input />
  </div>
  <Button
    form="productIdForm"
    txt="Search"
    type="submit"
  />
</form>
`;
